### PR TITLE
feat: `avm/res/data-protection/backup-vault`

### PR DESCRIPTION
**Pull request by segraef**
[avm.res.data-protection.backup-vault](https://github.com/segraef/bicep-registry-modules/actions/workflows/avm.res.data-protection.backup-vault.yml)

### CODE REVIEW

**Review by AlexanderSehr — CHANGES_REQUESTED** [review]
Requires uncommenting in the `/.github/CODEOWNERS` file

**Comment by segraef** [comment]
> Requires uncommenting in the `/.github/CODEOWNERS` file

done

File: .github/CODEOWNERS
Comment (AlexanderSehr) [inline .github/CODEOWNERS:40]: Team seems to be missing

Comment (eriqua) [reply .github/CODEOWNERS:40]: And added to parent team as per https://azure.github.io/Azure-Verified-Modules/specs/shared/#team-memberships

File: .github/workflows/avm.res.data-protection.backup-vault.yml
Comment (eriqua) [inline .github/workflows/avm.res.data-protection.backup-vault.yml:47]:
```suggestion
    runs-on: ubuntu-latest
```

File: .github/CODEOWNERS
Comment (segraef) [reply .github/CODEOWNERS:40]: Exist already:
@Azure/avm-res-consumption-budget-module-owners-bicep
@Azure/avm-res-consumption-budget-module-contributors-bicep

File: avm/res/data-protection/backup-vault/main.bicep
Comment (AlexanderSehr) [inline avm/res/data-protection/backup-vault/main.bicep:155]:
```suggestion
output systemAssignedMIPrincipalId string = backupVault.?identity.?principalId ?? ''
```

File: .github/CODEOWNERS
Comment (AlexanderSehr) [reply .github/CODEOWNERS:40]: May still be missing the owner assignment / its approval

Comment (segraef) [reply .github/CODEOWNERS:40]: both teams exist and parent teams are assigned

Comment (AlexanderSehr) [reply .github/CODEOWNERS:40]: True true

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**
